index.rst

```diff
@@ -85,6 +85,29 @@ When trying to launch a testcontainer from within a Docker container, e.g., in c
 1. The container has to provide a docker client installation. Either use an image that has docker pre-installed (e.g. the `official docker images <https://hub.docker.com/_/docker>`_) or install the client from within the `Dockerfile` specification.
 2. The container has to have access to the docker daemon which can be achieved by mounting `/var/run/docker.sock` or setting the `DOCKER_HOST` environment variable as part of your `docker run` command.
 
+Reusable Containers (Experimental)
+----------------------------------
+
+Containers can be reused across consecutive test runs.
+
+How to use?
+^^^^^^^^^^^
+
+1. Add `testcontainers.reuse.enable=true` to `~/.testcontainers.properties`
+2. Disable ryuk by setting the environment variable `TESTCONTAINERS_RYUK_DISABLED=true`
+3. Instantiate a container using `with_reuse`
+
+.. doctest::
+
+    >>> from testcontainers.core.container import DockerContainer
+
+    >>> with DockerContainer("hello-world").with_reuse() as container:
+    ...     first_id = container._container.id
+    >>> with DockerContainer("hello-world").with_reuse() as container:
+    ...     second_id == container._container.id
+    >>> print(first_id == second_id)
+    True
+
```

Comment:
Should the user be warned that by using this feature, containers need to be removed manually? (That this feature should not be used in a CI)

Also, do we need to make clear how this feature works (explaining the hash in use). -> If a container's run configuration changes, the hash changes and a new container will be used.

Reply:
seems like you have added these comments to the doc, i think that is fine. the hash would be great to add as users would benefit from knowing exactly what is hashed.

* `self.image,`
* `self._command,`
* `self.env,`
* `self.ports,`
* `self._name,`
* `self.volumes,`
* `str(tuple(sorted(self._kwargs.items()))),` - this may fail and why i want to have this be tucked away inside an obviously readable `if` block